### ISSUE TITLE
Yet another "There are no packages available for installation"

### ISSUE THREAD

**Issue by Asalle** [wbond/package_control#1141]
Hi there,

I used Package Control for a few weeks now and today it suddenly told me this:

```
Package Control: Error downloading channel. HTTP exception InvalidCertificateException (Host packagecontrol.io returned an invalid certificate ([SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:548))) downloading https://packagecontrol.io/channel_v3.json.
```

What should I do to bring my lovely Package Control back?

UPD: Well, it works with http://

**Comment by wbond** [comment]
Posting the debug log would be helpful.

Usually TLS (SSL) issues stem from:

 1. A proxy intercepting, decrypting and re-encrypting traffic
 2. An empty list of secure certificates
 3. Some sort of bug with handling an HTTPS proxy

The debug log helps identify what is likely the issue.

**Comment by Asalle** [comment]
```
ignored packages updated to: ["Theme - Midnight", "Vintage"]
reloading settings Packages/User/Package Control.sublime-settings
reloading settings Packages/User/Preferences.sublime-settings
Package Control: Error submitting usage information. HTTP exception InvalidCertificateException (Host packagecontrol.io returned an invalid certificate ([SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:548))) downloading https://packagecontrol.io/submit?version=1.0.2&sublime_version=3114&package=Theme+-+Midnight&sublime_platform=linux&operation=install&package_control_version=3.1.2.
reloading settings Packages/User/Package Control.sublime-settings
ignored packages updated to: ["Vintage"]
generating meta info summary
reloading settings Packages/User/Package Control.sublime-settings
reloading settings Packages/User/Preferences.sublime-settings
reloading settings Packages/User/Preferences.sublime-settings
reloading settings Packages/User/Preferences.sublime-settings
error: Error trying to parse settings: Unexpected character, expected a comma or closing bracket in Packages/User/Preferences.sublime-settings:5:2
reloading settings Packages/User/Preferences.sublime-settings
error: Error trying to parse settings: Unexpected character, expected a comma or closing bracket in Packages/User/Preferences.sublime-settings:5:2
reloading settings Packages/User/Preferences.sublime-settings
reloading settings Packages/User/Preferences.sublime-settings
error: Error trying to parse settings: Unexpected character, expected a comma or closing bracket in Packages/User/Preferences.sublime-settings:5:2
reloading settings Packages/User/Preferences.sublime-settings
reloading settings Packages/User/Preferences.sublime-settings
reloading settings Packages/User/Preferences.sublime-settings
reloading settings Packages/User/Preferences.sublime-settings
reloading settings Packages/User/Preferences.sublime-settings
indexing [queue 45]: no files were indexed out of the 1 queued, abandoning crawl
Unable to open /home/asalle/.config/sublime-text-3/Packages/Package Control/Package Control.sublime-settings
reloading settings Packages/User/Package Control.sublime-settings
reloading settings Packages/User/Package Control.sublime-settings
Package Control: Fetching list of available packages
  Platform: linux-x64
  Sublime Text Version: 3114
  Package Control Version: 3.1.2
Package Control: Download Debug
  URL: https://packagecontrol.io/channel_v3.json
  Timeout: 30
  Resolved IP: 50.116.33.29
  Resolved IPv6: 2600:3c02::f03c:91ff:fedf:6883
Package Control: Urllib Debug Proxy
  http_proxy: 
  https_proxy: 
  proxy_username: 
  proxy_password: 
Package Control: Found system CA bundle at /usr/lib/ssl/certs/ca-certificates.crt
Package Control: Urllib HTTPS Debug General
  Connecting to packagecontrol.io on port 443
Package Control: Urllib HTTPS Debug General
  Upgrading connection to SSL using CA certs file at /home/asalle/.config/sublime-text-3/Packages/User/Package Control.merged-ca-bundle
Package Control: Error downloading channel. HTTP exception InvalidCertificateException (Host packagecontrol.io returned an invalid certificate ([SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:548))) downloading https://packagecontrol.io/channel_v3.json.
```

Somehow it randomly works again.

**Comment by davij** [comment]
I am having the same problem on a mac. Here's the debug log: 

Package Control: Fetching list of available packages
  Platform: osx-x64
  Sublime Text Version: 3114
  Package Control Version: 3.1.2
Package Control: Download Debug
  URL: https://packagecontrol.io/channel_v3.json
  Timeout: 30
  Resolved IP: 50.116.33.29
  Resolved IPv6: 2600:3c02::f03c:91ff:fedf:6883
Package Control: Urllib Debug Proxy
  http_proxy: 
  https_proxy: 
  proxy_username: 
  proxy_password: 
Package Control: Found previously exported CA bundle at /Users/jdavila/Library/Application Support/Sublime Text 3/Packages/User/Package Control.system-ca-bundle
Package Control: Urllib HTTPS Debug General
  Connecting to packagecontrol.io on port 443
Package Control: Urllib HTTPS Debug General
  Upgrading connection to SSL using CA certs file at /Users/jdavila/Library/Application Support/Sublime Text 3/Packages/User/Package Control.merged-ca-bundle
Package Control: Error downloading channel. HTTP exception InvalidCertificateException (Host packagecontrol.io returned an invalid certificate ([SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:548))) downloading https://packagecontrol.io/channel_v3.json.

**Comment by akdor1154** [comment]
I am seeing this, also on Mac. My organization has a supposedly transparent SSL fudging proxy, however I have their root certificate set as trusted in the OS keychain, and OS X apps have no issues with it.

    curl --connect-timeout 5 -L -O -v --tlsv1 --cacert '/Users/jarrad/Library/Application Support/Sublime Text 3/Packages/User/Package Control.merged-ca-bundle' https://packagecontrol.io/channel_v3.json

works fine, and 

    curl --connect-timeout 5 -L -O -v --tlsv1 --cacert '/empty/certificate' https://packagecontrol.io/channel_v3.json

fails (suggesting that merged-ca-bundle indeed includes the required certificates, and something else must be going on).

**Comment by wbond** [comment]
@akdor1154 Can you provide some verbose output from curl for the successful connection? I am trying to find something to work from to resolve the issue. There are multiple reports of this type of issue, but I haven't been able to identify at what layer the issue is happening.

**Comment by chshouyu** [comment]
#1187

**Comment by rkd25** [comment]
Hi, 

I am facing this problem There are no packages available for installation. Please get resolved. 


DPI scale: 1
startup, version: 3126 windows x64 channel: stable
executable: /C/Program Files/Sublime Text 3/sublime_text.exe
working dir: /C/Program Files/Sublime Text 3
packages path: /C/Users/rakesh.dubey/AppData/Roaming/Sublime Text 3/Packages
state path: /C/Users/rakesh.dubey/AppData/Roaming/Sublime Text 3/Local
zip path: /C/Program Files/Sublime Text 3/Packages
zip path: /C/Users/rakesh.dubey/AppData/Roaming/Sublime Text 3/Installed Packages
ignored_packages: ["Vintage"]
pre session restore time: 0.349038
startup time: 0.479038
first paint time: 0.479038
first paint time: 0.479038
reloading plugin Default.auto_indent_tag
reloading plugin Default.block
reloading plugin Default.comment
reloading plugin Default.convert_syntax
reloading plugin Default.copy_path
reloading plugin Default.delete_word
reloading plugin Default.detect_indentation
reloading plugin Default.duplicate_line
reloading plugin Default.echo
reloading plugin Default.exec
reloading plugin Default.fold
reloading plugin Default.font
reloading plugin Default.goto_line
reloading plugin Default.history_list
reloading plugin Default.indentation
reloading plugin Default.install_package_control
reloading plugin Default.kill_ring
reloading plugin Default.mark
reloading plugin Default.new_templates
reloading plugin Default.open_context_url
reloading plugin Default.open_in_browser
reloading plugin Default.pane
reloading plugin Default.paragraph
reloading plugin Default.paste_from_history
reloading plugin Default.profile
reloading plugin Default.quick_panel
reloading plugin Default.run_syntax_tests
reloading plugin Default.save_on_focus_lost
reloading plugin Default.scroll
reloading plugin Default.set_unsaved_view_name
reloading plugin Default.settings
reloading plugin Default.show_scope_name
reloading plugin Default.side_bar
reloading plugin Default.sort
reloading plugin Default.swap_line
reloading plugin Default.switch_file
reloading plugin Default.symbol
reloading plugin Default.transform
reloading plugin Default.transpose
reloading plugin Default.trim_trailing_white_space
reloading plugin CSS.css_completions
reloading plugin Diff.diff
reloading plugin HTML.encode_html_entities
reloading plugin HTML.html_completions
reloading plugin 0_package_control_loader.00-package_control
reloading plugin 0_package_control_loader.02-bz2
reloading plugin Package Control.1_reloader
reloading plugin Package Control.2_bootstrap
reloading plugin Package Control.Package Control
plugins loaded
Package Control: Fetching list of available packages and dependencies
  Platform: windows-x64
  Sublime Text Version: 3126
  Package Control Version: 3.2.1
Package Control: Fetching list of available packages and dependencies
  Platform: windows-x64
  Sublime Text Version: 3126
  Package Control Version: 3.2.1
Package Control: No updated packages
Package Control: Fetching list of available packages and dependencies
  Platform: windows-x64
  Sublime Text Version: 3126
  Package Control Version: 3.2.1
>>> import urllib.request,os,hashlib; h = 'df21e130d211cfc94d9b0905775a7c0f' + '1e3d39e33b79698005270310898eea76'; pf = 'Package Control.sublime-package'; ipp = sublime.installed_packages_path(); urllib.request.install_opener( urllib.request.build_opener( urllib.request.ProxyHandler()) ); by = urllib.request.urlopen( 'http://packagecontrol.io/' + pf.replace(' ', '%20')).read(); dh = hashlib.sha256(by).hexdigest(); print('Error validating download (got %s instead of %s), please try manual install' % (dh, h)) if dh != h else open(os.path.join( ipp, pf), 'wb' ).write(by)
275309
Package Control: Fetching list of available packages and dependencies
  Platform: windows-x64
  Sublime Text Version: 3126
  Package Control Version: 3.2.1
>>> import urllib.request,os,hashlib; h = 'df21e130d211cfc94d9b0905775a7c0f' + '1e3d39e33b79698005270310898eea76'; pf = 'Package Control.sublime-package'; ipp = sublime.installed_packages_path(); urllib.request.install_opener( urllib.request.build_opener( urllib.request.ProxyHandler()) ); by = urllib.request.urlopen( 'http://packagecontrol.io/' + pf.replace(' ', '%20')).read(); dh = hashlib.sha256(by).hexdigest(); print('Error validating download (got %s instead of %s), please try manual install' % (dh, h)) if dh != h else open(os.path.join( ipp, pf), 'wb' ).write(by)
275309
>>> import urllib.request,os,hashlib; h = 'df21e130d211cfc94d9b0905775a7c0f' + '1e3d39e33b79698005270310898eea76'; pf = 'Package Control.sublime-package'; ipp = sublime.installed_packages_path(); urllib.request.install_opener( urllib.request.build_opener( urllib.request.ProxyHandler()) ); by = urllib.request.urlopen( 'http://packagecontrol.io/' + pf.replace(' ', '%20')).read(); dh = hashlib.sha256(by).hexdigest(); print('Error validating download (got %s instead of %s), please try manual install' % (dh, h)) if dh != h else open(os.path.join( ipp, pf), 'wb' ).write(by)
275309
Package Control: Fetching list of available packages and dependencies
  Platform: windows-x64
  Sublime Text Version: 3126
  Package Control Version: 3.2.1
ignored packages updated to: ["Package Control", "Vintage"]
unloading plugin Package Control.1_reloader
unloading plugin Package Control.2_bootstrap
unloading plugin Package Control.Package Control
reloading settings Packages/Package Control/Package Control.sublime-settings
reloading settings Packages/User/Package Control.sublime-settings
reloading settings Packages/User/Preferences.sublime-settings
Package Control: Download Debug
  URL: https://packagecontrol.io/submit?version=3.2.1&sublime_platform=windows&package_control_version=None&package=Package+Control&sublime_version=3126&operation=remove
  Timeout: 30
  Resolved IP: 50.116.33.29
Package Control: WinINet Debug Proxy
  proxy: 
  proxy bypass: 
  proxy username: 
  proxy password: 
Package Control: WinINet HTTPS Debug General
  Server SSL Certificate:
    subject: US, Massachusetts, Private Organization, 471714639, US, Massachusetts, Newbury, Codex Non Sufficit LC, codexns.io
    issuer: US, GeoTrust Inc., GeoTrust EV SSL CA - G4
    common name: codexns.io
    issue date: Mon, 28 Mar 2016 05:30:00 GMT
    expire date: Mon, 28 Nov 2016 05:29:59 GMT
Package Control: WinINet HTTPS Debug Write
  GET /submit?version=3.2.1&sublime_platform=windows&package_control_version=None&package=Package+Control&sublime_version=3126&operation=remove HTTP/1.1
  User-Agent: Package Control v3.2.1
  Host: packagecontrol.io
  Connection: Keep-Alive
  Cache-Control: no-cache
  Accept-Encoding: bzip2,gzip,deflate
Package Control: WinINet HTTPS Debug Read
  HTTP/1.1 200 OK
  Server: nginx
  Date: Sat, 12 Nov 2016 17:20:29 GMT
  Content-Type: application/json
  Transfer-Encoding: chunked
  Connection: keep-alive
  Vary: Accept-Encoding
  X-App-Version: 1.3.3
  Strict-Transport-Security: max-age=15552000
  Content-Encoding: gzip
Package Control: The package Package Control has been removed
reloading settings Packages/User/Package Control.sublime-settings
ignored packages updated to: ["Vintage"]
reloading settings Packages/User/Package Control.sublime-settings
reloading settings Packages/User/Preferences.sublime-settings
reloading settings Packages/User/Preferences.sublime-settings
Package Control: WinINet HTTPS Debug General
  Closing connection to packagecontrol.io on port 443 after 1 request
reloading plugin Package Control.1_reloader
reloading plugin Package Control.2_bootstrap
reloading plugin Package Control.Package Control
reloading settings Packages/Package Control/Package Control.sublime-settings
Package Control: Settings change detected, clearing cache
Package Control: Download Debug
  URL: https://packagecontrol.io/submit?version=3.2.1&sublime_platform=windows&package_control_version=3.2.1&package=Package+Control&sublime_version=3126&operation=install
  Timeout: 30
  Resolved IP: 50.116.33.29
Package Control: WinINet Debug Proxy
  proxy: 
  proxy bypass: 
  proxy username: 
  proxy password: 
Package Control: WinINet HTTPS Debug General
  Server SSL Certificate:
    subject: US, Massachusetts, Private Organization, 471714639, US, Massachusetts, Newbury, Codex Non Sufficit LC, codexns.io
    issuer: US, GeoTrust Inc., GeoTrust EV SSL CA - G4
    common name: codexns.io
    issue date: Mon, 28 Mar 2016 05:30:00 GMT
    expire date: Mon, 28 Nov 2016 05:29:59 GMT
Package Control: WinINet HTTPS Debug Write
  GET /submit?version=3.2.1&sublime_platform=windows&package_control_version=3.2.1&package=Package+Control&sublime_version=3126&operation=install HTTP/1.1
  User-Agent: Package Control v3.2.1
  Host: packagecontrol.io
  Connection: Keep-Alive
  Cache-Control: no-cache
  Accept-Encoding: bzip2,gzip,deflate
Package Control: WinINet HTTPS Debug Read
  HTTP/1.1 200 OK
  Server: nginx
  Date: Sat, 12 Nov 2016 17:21:13 GMT
  Content-Type: application/json
  Transfer-Encoding: chunked
  Connection: keep-alive
  Vary: Accept-Encoding
  X-App-Version: 1.3.3
  Strict-Transport-Security: max-age=15552000
  Content-Encoding: gzip
Package Control: Skipping automatic upgrade, last run at 2016-11-12 22:37:18, next run at 2016-11-12 23:37:18 or after
reloading settings Packages/User/Package Control.sublime-settings
Package Control: WinINet HTTPS Debug General
  Closing connection to packagecontrol.io on port 443 after 1 request
{}
Package Control: Fetching list of available packages and dependencies
  Platform: windows-x64
  Sublime Text Version: 3126
  Package Control Version: 3.2.1
Package Control: Fetching list of available packages and dependencies
  Platform: windows-x64
  Sublime Text Version: 3126
  Package Control Version: 3.2.1
Package Control: Fetching list of available packages and dependencies
  Platform: windows-x64
  Sublime Text Version: 3126
  Package Control Version: 3.2.1
Unable to open /C/Users/rakesh.dubey/AppData/Roaming/Sublime Text 3/Packages/Package Control/Package Control.sublime-settings
Package Control: Fetching list of available packages and dependencies
  Platform: windows-x64
  Sublime Text Version: 3126
  Package Control Version: 3.2.1
>>> import urllib.request,os,hashlib; h = 'df21e130d211cfc94d9b0905775a7c0f' + '1e3d39e33b79698005270310898eea76'; pf = 'Package Control.sublime-package'; ipp = sublime.installed_packages_path(); urllib.request.install_opener( urllib.request.build_opener( urllib.request.ProxyHandler()) ); by = urllib.request.urlopen( 'http://packagecontrol.io/' + pf.replace(' ', '%20')).read(); dh = hashlib.sha256(by).hexdigest(); print('Error validating download (got %s instead of %s), please try manual install' % (dh, h)) if dh != h else open(os.path.join( ipp, pf), 'wb' ).write(by)
275309
Package Control: Fetching list of available packages and dependencies
  Platform: windows-x64
  Sublime Text Version: 3126
  Package Control Version: 3.2.1
>>> import urllib.request,os,hashlib; h = 'df21e130d211cfc94d9b0905775a7c0f' + '1e3d39e33b79698005270310898eea76'; pf = 'Package Control.sublime-package'; ipp = sublime.installed_packages_path(); urllib.request.install_opener( urllib.request.build_opener( urllib.request.ProxyHandler()) ); by = urllib.request.urlopen( 'http://packagecontrol.io/' + pf.replace(' ', '%20')).read(); dh = hashlib.sha256(by).hexdigest(); print('Error validating download (got %s instead of %s), please try manual install' % (dh, h)) if dh != h else open(os.path.join( ipp, pf), 'wb' ).write(by)
275309
Package Control: Fetching list of available packages and dependencies
  Platform: windows-x64
  Sublime Text Version: 3126
  Package Control Version: 3.2.1
{}
Package Control: Fetching list of available packages and dependencies
  Platform: windows-x64
  Sublime Text Version: 3126
  Package Control Version: 3.2.1
Package Control: Fetching list of available packages and dependencies
  Platform: windows-x64
  Sublime Text Version: 3126
  Package Control Version: 3.2.1
Package Control: Fetching list of available packages and dependencies
  Platform: windows-x64
  Sublime Text Version: 3126
  Package Control Version: 3.2.1

**Comment by wbond** [comment]
Please try manually upgrading to Package Control 3.3.0. This should hopefully resolve the invalid certificate error.
